API incompatibility between webhook/apiserver and k8s.io/apiserver

**Is your feature request related to a problem? Please describe.**

Due to https://github.com/kubernetes/apiserver/commit/2bba9591a36aa330ad84b092be8fb8d92de4bcd7, building with mainline k8s libraries (suddenly necessary thanks to https://github.com/kubernetes/kubernetes/issues/106536 and https://github.com/kubernetes/component-base/issues/5) now fails with

```
$ go build
# github.com/cert-manager/cert-manager/pkg/acme/webhook/apiserver
../../go/pkg/mod/github.com/cert-manager/cert-manager@v1.9.1/pkg/acme/webhook/apiserver/apiserver.go:162:35: cannot use challengeHandler (variable of type *challengepayload.REST) as type "k8s.io/apiserver/pkg/registry/rest".Storage in assignment:
        *challengepayload.REST does not implement "k8s.io/apiserver/pkg/registry/rest".Storage (missing Destroy method)
```

**Describe the solution you'd like**
cert-manager should implement the new method in -I think- https://github.com/cert-manager/cert-manager/blob/231a0460b40e18537843037cbf083c9ab1cdfc2f/pkg/acme/webhook/registry/challengepayload/challenge_payload.go

**Describe alternatives you've considered**
N/A

**Additional context**
Was a bit of a pita dealing with the otel changes breaking k8s builds breaking cert-manager webhook builds...

/kind feature

Listed it as "feature" because it's not quite a "bug". But it's not really a feature either, admittedly.

--- a/pkg/acme/webhook/registry/challengepayload/challenge_payload.go
+++ b/pkg/acme/webhook/registry/challengepayload/challenge_payload.go
@@ -100,3 +100,6 @@ func (r *REST) callSolver(req v1alpha1.ChallengeRequest) (v1alpha1.ChallengeResp
 		},
 	}, nil
 }
+
+func (r *REST) Destroy() {
+}

--- a/pkg/acme/webhook/registry/challengepayload/challenge_payload.go
+++ b/pkg/acme/webhook/registry/challengepayload/challenge_payload.go
@@ -101,5 +101,9 @@ func (r *REST) callSolver(req v1alpha1.ChallengeRequest) (v1alpha1.ChallengeResp
 	}, nil
 }
 
+// This resource type isn't actually persisted anywhere, it is only submitted to the
+// DNS01 solver webhooks, so there's nothing to do to delete a resource/it doesn't
+// make sense in this context.
+// see: https://github.com/cert-manager/cert-manager/pull/5346#discussion_r959521656
 func (r *REST) Destroy() {
 }

